### DISCUSSION TITLE
PartDB and Authentik

### DISCUSSION THREAD

**Discussion by pixbalance** [goauthentik/authentik#9290] [Q&A]
Is anybody using PartDB with Authenik for SSO and SAML? I tried to transfer the information from Keycloak to Authentik (https://docs.part-db.de/installation/saml_sso.html)

I have partdb, authentik and NGINX installed on unraid as docker.
My domain is https://partdb.mydomain.de, Login with local credentials in PartDB is working.

**Current status:**
When I click on the SSO login button on https://partdb.mydomain.de I end up in a error 500 page.
When I open Authentik https://auth.mydomain.de, log in and click on the App Icon for PartDB, login is working and a user is created.
Although I have added the user attributes email, lastName and firstName the user is created with a random number and no information used.
When I am logged in into PartDB and try to logout, I end up at error 500 page.

I have made the following settings:

**SAML_ENABLED:**
1

**SAML_SP_ENTITY_ID:**
https://partdb.mydomain.de
this value is from the provider metadata attribute "entityID"

**SAML_IDP_ENTITY_ID:**
https://partdb.mydomain.de
same as SAML_SP_ENTITY_ID

**SAML_IDP_SINGLE_SIGN_ON_SERVICE:**
https://partdb.mydomain.de/application/saml/partdb/sso/binding/post/
this value is from the provider metadata attribute SingleLogoutService -> Location with Binding Http-Post

**SAML_IDP_SINGLE_LOGOUT_SERVICE:**
https://partdb.mydomain.de/application/saml/partdb/slo/binding/post/
this value is from the provider metadata attribute SingleSignOnService -> Location with Binding Http-Post

**SAML_IDP_X509_CERT:**
....
this value is from the provider metadata attribute ds:X509Certificate

**DEFAULT_URI:**
https://partdb.mydomain.de/

**SAML_ROLE_MAPPING:**
{"admin": 1,  "family": 5, "guest": 4, "*": 2}

**Comment by alextrical** [comment]
I'm in the same boat.
I have partdb, authentik and traefik installed on unraid via docker containers.

I have also made the following settings:

SAML_ENABLED:1

SAML_SP_ENTITY_ID:https://partdb.mydomain.de
this value is from the Authentik WebUI SAML Provider Overview "EntityID/Issuer"

SAML_IDP_ENTITY_ID:https://partdb.mydomain.de
same as SAML_SP_ENTITY_ID

SAML_IDP_SINGLE_SIGN_ON_SERVICE:https://partdb.mydomain.de/application/saml/partdb/sso/binding/post/
this value is from the Authentik WebUI SAML Provider Overview "SSO URL (Post)"

SAML_IDP_SINGLE_LOGOUT_SERVICE:https://partdb.mydomain.de/application/saml/partdb/slo/binding/post/
this value is from the Authentik WebUI SAML Provider Overview "SLO URL (Post)"

SAML_IDP_X509_CERT:
....
this value is from the provider metadata attribute ds:X509Certificate

DEFAULT_URI:https://partdb.mydomain.de/
same as SAML_SP_ENTITY_ID but with a trailing /

SAML_ROLE_MAPPING:{ "*": 2, "admin": 1, "editor": 3}


With this setup i get the following error in the logs:
```
[2026-02-27T10:02:08.404103+00:00] php.ERROR: Warning: openssl_sign(): Supplied key param cannot be coerced into a private key {"exception":"[object] (ErrorException(code: 0): Warning: openssl_sign(): Supplied key param cannot be coerced into a private key at /var/www/html/vendor/robrichards/xmlseclibs/src/XMLSecurityKey.php:563)
[stacktrace]
#0 /var/www/html/vendor/robrichards/xmlseclibs/src/XMLSecurityKey.php(644): RobRichards\\XMLSecLibs\\XMLSecurityKey->signOpenSSL()
#1 /var/www/html/vendor/onelogin/php-saml/src/Saml2/Auth.php(789): RobRichards\\XMLSecLibs\\XMLSecurityKey->signData()
#2 /var/www/html/vendor/onelogin/php-saml/src/Saml2/Auth.php(726): OneLogin\\Saml2\\Auth->buildMessageSignature()
#3 /var/www/html/vendor/onelogin/php-saml/src/Saml2/Auth.php(560): OneLogin\\Saml2\\Auth->buildRequestSignature()
#4 /var/www/html/vendor/nbgrp/onelogin-saml-bundle/src/Controller/Login.php(66): OneLogin\\Saml2\\Auth->login()
#5 /var/www/html/vendor/nbgrp/onelogin-saml-bundle/src/Controller/Login.php(48): Nbgrp\\OneloginSamlBundle\\Controller\\Login->processLoginAndGetRedirectUrl()
#6 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(183): Nbgrp\\OneloginSamlBundle\\Controller\\Login->__invoke()
#7 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(76): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#8 /var/www/html/vendor/symfony/http-kernel/Kernel.php(193): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#9 /var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php(35): Symfony\\Component\\HttpKernel\\Kernel->handle()
#10 /var/www/html/vendor/autoload_runtime.php(32): Symfony\\Component\\Runtime\\Runner\\Symfony\\HttpKernelRunner->run()
#11 /var/www/html/public/index.php(5): require_once()
#12 {main}
"} {"url":"/saml/login","ip":"172.18.0.19","http_method":"GET","server":"partdb.mydomain.de","referrer":"https://partdb.mydomain.de/en/login?_target_path=%2F","token":null}
[2026-02-27T10:02:08.409992+00:00] request.CRITICAL: Uncaught PHP Exception Exception: "Failure Signing Data: error:1E08010C:DECODER routines::unsupported - SHA256" at XMLSecurityKey.php line 564 {"exception":"[object] (Exception(code: 0): Failure Signing Data: error:1E08010C:DECODER routines::unsupported - SHA256 at /var/www/html/vendor/robrichards/xmlseclibs/src/XMLSecurityKey.php:564)
[stacktrace]
```

keys generated on a different machine using
```
openssl genrsa -out rsa_key.p8 2048
openssl req -new -x509 -key rsa_key.p8 -out sp.crt -days 3650 -subj "/CN=partdb.mydomain.de" -addext "subjectAltName=DNS:partdb.mydomain.de"
 ```
 
SAML_SP_PRIVATE_KEY:"MIIE..."
 ```
cat rsa_key.p8 | grep -v '^---' | tr -d '\n\r' | sed 's/$/=/'
 ```
 
SAML_SP_X509_CERT:"MIID..."
 ```
cat sp.crt | grep -v '^---' | tr -d '\n\r' | sed 's/$/=/'
 ```
